tests/job/test_dynamic.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+import json
+import os
+from pyiron_base import Project
+from pyiron_base.jobs.dynamic import create_new_job_type
+
+
+def write_input(input_dict, working_directory="."):
+    with open(os.path.join(working_directory, "input.json"), "w") as f:
+        json.dump(input_dict, f)
+
+
+def collect_output(working_directory="."):
+    with open(os.path.join(working_directory, "output.json"), "r") as f:
+        return json.load(f)
+
+
+class DynamicTest(unittest.TestCase):
```

Comment:
`TestWithProject` could be helpful here.

Reply:
That's a good idea - I updated the test to use the `TestWithProject` class.